packages/relay/src/lib/eth.ts

```diff
@@ -1211,6 +1211,25 @@ export class EthImpl implements Eth {
       ]);
       if (result) {
         if (result?.type === constants.TYPE_TOKEN) {
+          if (blockNumber && !this.common.blockTagIsLatestOrPending(blockNumber)) {
+            const blockNumberInt = parseInt(blockNumber, 16);
+            const blockInfo = await this.mirrorNodeClient.getBlock(blockNumberInt, requestDetails);
+
+            if (!blockInfo) {
+              throw predefined.UNKNOWN_BLOCK(`Block number ${blockNumber} does not exist`);
+            }
+
+            const tokenId = Utils.addressToTokenId(address);
+            const tokenInfo = await this.mirrorNodeClient.getTokenById(tokenId, requestDetails);
```

Comment:
can we use the `timestamp` argument in the `tokens/{tokenId}` endpoint? this way we avoid having the logic below to compare timestamps.